float64[3] start_point
float64[3] end_point
float64[3] ref_point
---
bool success
